lib/agent.js

```diff
@@ -377,6 +377,7 @@ Agent.prototype.request = function request(url) {
   // Start multicast monitoring timer in case of multicast request
   if (url.multicast === true) {
     req.multicastTimer = setTimeout(function() {
+      delete that._tkToReq[req._packet.token.readUInt32BE(0)]
```

Comment:
Noticed that `req._packet.token` is undefined if `end()` is never called on the response. Once the timeout function is invoked, `readUInt32BE()` is called on the `undefined` token and an exception is thrown.